### PR TITLE
Iteration tests

### PR DESCRIPTION
**Pull request by bramtayl**

### CODE REVIEW

File: src/abstractdataframe/iteration.jl [hunk 1]
```diff
@@ -19,9 +19,9 @@ which is a view that acts like a one-row DataFrame.
 eachrow(df::AbstractDataFrame) = DFRowIterator(df)
 
 Base.start(itr::DFRowIterator) = 1
-Base.done(itr::DFRowIterator, i::Int) = i > size(itr.df, 1)
+Base.done(itr::DFRowIterator, i::Int) = i > length(itr)
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]: Please keep this as-is, there have been discussions about deprecating `length` altogether.

File: src/abstractdataframe/iteration.jl [hunk 2]
```diff
@@ -33,9 +33,9 @@ end
 eachcol(df::AbstractDataFrame) = DFColumnIterator(df)
 
 Base.start(itr::DFColumnIterator) = 1
-Base.done(itr::DFColumnIterator, j::Int) = j > size(itr.df, 2)
+Base.done(itr::DFColumnIterator, j::Int) = j > length
```
Comment (ararslan) [inline src/abstractdataframe/iteration.jl]: This doesn't make sense

File: src/abstractdataframe/iteration.jl [hunk 3]
```diff
@@ -33,9 +33,9 @@ end
...
 Base.next(itr::DFColumnIterator, j::Int) = ((_names(itr.df)[j], itr.df[j]), j + 1)
-Base.size(itr::DFColumnIterator) = (size(itr.df, 2), )
+Base.size(itr::DFColumnIterator) = (length(itr), )
```
Comment (ararslan) [inline src/abstractdataframe/iteration.jl]: This change should also be reverted

**Comment by bramtayl** [comment]
I saw the coverage at 74% and I was bored and I thought maybe it would be helpful if I added a couple tests. I'm planning to make a whole bunch more test PRs if that would be helpful.

**Comment by nalimilan** [comment]
Sure. Improving coverage is always helpful!

**Pull request closed without merging**